examples/5_secret_storage.py

```diff
@@ -0,0 +1,419 @@
+            team, self.NAMESPACE, tool, info, expected_revision
+        )
+        return res
+
+    async def list_tools(self, team) -> List[str]:
+        res = await self.kvstore.list_entrykeys(team, self.NAMESPACE)
+        keys = (
+            [e.entry_key for e in res.entry_keys if not e.entry_key.startswith("_")]
+            if res.entry_keys
+            else []
+        )
+        return keys
+
+
+async def rental_user(bot, rental, team, username):
+    res = await rental.list_tools(team)
+    print(res)
+
+    tool = "laz0rs"
+    res = await rental.lookup(team, tool)
+    print("LOOKUP: ", res)
+
+    res = await rental.add(team, tool)
+    print("ADD: ", res)
+
+    res = await rental.remove(team, tool)
+    print("REMOVE: ", res)
+
+    res = await rental.add(team, tool)
+    print("ADD: ", res)
+
+    res = await rental.update_reservation(
+        team, username, tool, "2044-03-12", reserve=True
+    )
+    print("RESERVE: ", res)
+
+    res = await rental.update_reservation(
+        team, username, tool, "2044-06-12", reserve=True
+    )
+    print("RESERVE: ", res)
+
+    res = await rental.lookup(team, tool)
+    print("LOOKUP: ", res)
+
+    res = await rental.update_reservation(
+        team, username, tool, "2044-06-12", reserve=False
+    )
+    print("UNRESERVE: ", res)
+
+    res = await rental.lookup(team, tool)
+    print("LOOKUP: ", res)
+
+
+async def concurrent_rental_users(bot, rental, team, username):
+    tool = "time travel machine"
+
+    async def concurrent_rental_user(user_id: int):
+        date = "2044-10-0{}".format(user_id)
+        user = "user{}".format(user_id)
+
+        i = 0
+        while True:
+            # keep trying to reserve for user's unique date until successful
+            res = await rental.update_reservation(team, user, tool, date, reserve=True)
+            i += 1
+            print("{}, attempt {}, TRY TO RESERVE: {}".format(user, i, res))
+            if type(res) == keybase1.KVPutResult:
+                return
+
+    async def pre():
+        while True:
+            res = await rental.remove("yourhackerspace", tool)
+            if type(res) == keybase1.KVDeleteEntryResult:
+                print("REMOVE: {}".format(res))
+                break
+
+    async def post():
+        # check that the tool has been reserved for all 5 unique dates
+        res = await rental.lookup("yourhackerspace", tool)
+        print("LOOKUP: {}".format(res))
+        assert len(json.loads(res.entry_value)) == 6  # one key is "_key"
+
+    await asyncio.wait_for(pre(), timeout=5.0)
+    # have 5 users concurrently try to reserve the same tool for 5 unique dates
+    await asyncio.gather(
+        concurrent_rental_user(1),
+        concurrent_rental_user(2),
+        concurrent_rental_user(3),
+        concurrent_rental_user(4),
+        concurrent_rental_user(5),
+    )
+    await post()
+
+
+async def main():
+    print("Starting 5_secret_storage example...")
+
+    team = "yourhackerspace"
```

Comment:
can we make this a file-level constant and reference it in all the other places?